iothub_client/src/iothub_client_core.c

```diff
@@ -1691,13 +1695,20 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_SetOption(IOTHUB_CLIENT_CORE_HANDLE iotHub
         }
         else
         {
-            /*Codes_SRS_IOTHUBCLIENT_02_038: [If optionName doesn't match one of the options handled by this module then IoTHubClient_SetOption shall call IoTHubClientCore_LL_SetOption passing the same parameters and return what IoTHubClientCore_LL_SetOption returns.] */
-            result = IoTHubClientCore_LL_SetOption(iotHubClientInstance->IoTHubClientLLHandle, optionName, value);
-            if (result != IOTHUB_CLIENT_OK)
+            if (strcmp(OPTION_CONVENIENCE_LOOP_TIME, optionName) == 0)
             {
```

Comment:
You should define if there is a range of valid values, and verify it here (and fail setoption of the number is out of range).
TODO: discuss with team what a good range would be .

Reply:
setting the upperlimit to this to be 100ms